### PR TITLE
Add reducers and actions

### PR DESCRIPTION
**Pull request by erickma1**
In this milestone I did the following;

- Add Redux Toolkit 
- Create a directory that will contain all your Redux logic
- Configure a Redux store

### CODE REVIEW

File: src/redux/configureStore.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import categoriesReducer from './categories/categories';
+
+const reducer = combineReducers({
+  booksReducer,
+  categoriesReducer,
+});
+
+const store = createStore(reducer);
+
+export default store;
```
Comment (anagudelogu) [inline src/redux/configureStore.js]: - [ ] You did a great job working on this milestone! 💯 However, please consider using the recommended approach by Redux to set up your store which is using `configureStore` from `@reduxjs/toolkit` 🤓 🔧

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  type: ADD_BOOK,
+  payload,
+});
+
+export const removeBook = (id) => ({
+  type: REMOVE_BOOK,
+  id,
+});
+
+export default booksReducer;
```
Comment (anagudelogu) [inline src/redux/books/books.js]: - [ ] According to the requirements:

image

Please consider using the `@reduxjs/toolkit` library to create slices using the `createSlice` method as this is the updated and recommended way of defining slices! 🤓 🔧

File: src/redux/categories/categories.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    default:
+      return state;
+  }
+};
+
+export const checkStatus = () => ({
+  type: CHECK_STATUS,
+});
+
+export default categoriesReducer;
```
Comment (anagudelogu) [inline src/redux/categories/categories.js]: - [ ] The same happens here with your categories:

image

Please consider using `createSlice`! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @erickma1, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by erickma1** [comment]
set up your store which is using configureStore from @reduxjs/toolkit
create slices using the createSlice method

File: src/redux/books/booksSlice.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+const initialState = {
+  books: [],
+};
+const booksSlice = createSlice({
+  name: 'addBooks',
+  initialState,
+  reducers: {
+    addBooks(state, action) {
+      state.push(action.payload);
+    },
```
Comment (Qoosim) [inline src/redux/books/booksSlice.js]: - [ ] Great job implementing required changes from the previous reviewer. However, there's a little stuff to look into here. There's `books` array of object inside your state which is equivalent to initialState. Pushing  the book object into state directly will not fill the book array of objects. Baiscally it should be `state.books.push(action.payload)`

File: src/redux/books/booksSlice.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+    delBooks(state, action) {
+      state.filter((book) => book.id !== action.payload);
+    },
+  },
```
Comment (Qoosim) [inline src/redux/books/booksSlice.js]: - [ ] Same correction should be made here too

``` 
  state.books.filter()
```

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle: 

Hi @erickma1, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Highlights
-  Implemented some required changes. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [OPTIONAL] Kindly add some future features under `Future Features` heading in your README file. It is empty at this moment.

futre
 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by erickma1** [comment]
Corrected pushing the book object into state directly
Added future features to README

**Review by V-Blaze — APPROVED** [review]
Hi @erickma1  , 

# Approved 🟢 🟢 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy

## Highlights

- Installed and setup redux with `@reduxjs/toolkit`💯
-  BookSLice implemented correctly 👍
-  Nice communication with the reviewer🟢 👍

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**